diff --git a/ui/input-linux.c b/ui/input-linux.c
index 05c0c98..7f308b8 100644
--- a/ui/input-linux.c
+++ b/ui/input-linux.c
@@ -65,6 +65,7 @@ struct InputLinux {
     QTAILQ_ENTRY(InputLinux) next;
 };

+IOThread *g_iothread = NULL;

 static QTAILQ_HEAD(, InputLinux) inputs = QTAILQ_HEAD_INITIALIZER(inputs);

@@ -311,6 +312,10 @@ static void input_linux_complete(UserCreatable *uc, Error **errp)
         return;
     }

+    if (!g_iothread) {
+        g_iothread = iothread_create("meme_iothread", errp);
+    }
+
     il->fd = open(il->evdev, O_RDWR);
     if (il->fd < 0)  {
         error_setg_file_open(errp, errno, il->evdev);
@@ -393,7 +398,8 @@ static void input_linux_complete(UserCreatable *uc, Error **errp)
         }
     }

-    qemu_set_fd_handler(il->fd, input_linux_event, NULL, il);
+    aio_set_fd_handler(iothread_get_aio_context(g_iothread), il->fd, false,
+            input_linux_event, NULL, NULL, NULL, il);
     if (il->keycount) {
         /* delay grab until all keys are released */
         il->grab_request = true;
