diff --git a/lib/vcs_info.zsh b/lib/vcs_info.zsh
index e60938c..88acd1c 100644
--- a/lib/vcs_info.zsh
+++ b/lib/vcs_info.zsh
@@ -1,3 +1,7 @@
+autoload -Uz is-at-least
+if is-at-least 5.8.1 || ! is-at-least 5.0.3; then
+  return
+fi
 # Don't skip this file until a Zsh release does the necessary quoting.
 # This is because even though 5.8.1 undid recursive prompt_subst inside
 # prompt sequences, % characters in relevant fields will still be rendered
